src/index.ts

```diff
@@ -154,44 +145,31 @@ function executeQuery(
       return done(null)
     }
 
+    const elementIdAttributeName = 'data-wdio-testing-lib-element-id';
+    const makeSelectorResult = (element: HTMLElement) => {
+      let elementId = element.getAttribute(elementIdAttributeName);
+      if (elementId == null) {
+        const idSequence = window.TestingLibraryDom.idSequence ?? 10000;
+        window.TestingLibraryDom.idSequence = idSequence + 1;
+        elementId = `${idSequence}`;
+        element.setAttribute(elementIdAttributeName, elementId)
+      }
+      return ({selector: `[${elementIdAttributeName}="${elementId}"]`})
+    };
+
     if (Array.isArray(result)) {
-      return done(
-        result.map((element) => ({
-          selector: window.Simmer(element),
-          element,
-        })),
-      )
+      return done(result.map(makeSelectorResult));
     }
 
-    return done({
-      selector: window.Simmer(result),
-      element: result,
-    })
+    return done(makeSelectorResult(result));
```

Comment:
I like the idea of generating a selector rather than returning it directly, when returning directly it's not possible to refetch the element so this is an improvement on that. There is one area where Simmer is better though and that is when an element is recreated: in that case the attribute that we have added is lost and we can no longer refetch it. This happens when a React element's `key` changes, and I imagine it is relatively common in other JS frameworks also.

Another thing I think would be worthwhile changing is to use a random number rather than incrementing `idSequence`, that way we don't need to store anything on `TestingLibraryDom` or worry about unexpected behaviour. For example what happens if we accidentally execute the DOM Testing Library script again, or call `configure`? I assume it would be fine but I'm not sure.

Anyways here is a version of the `makeSelectorResult` helper I was playing with that uses Simmer first and then falls back to a generated Id:
```typescript
  function makeSelectorResult(element: HTMLElement) {
    // use simmer if possible to allow element refetching by position, otherwise
    // situations such as a React key change causes refetching to fail.
    const selector = window.Simmer(element)
    if (selector) return {selector}

    // use generated element id as selector if Simmer fails
    const elementIdAttributeName = 'data-wdio-testing-lib-element-id'
    let elementId = element.getAttribute(elementIdAttributeName)

    // if id doesn't already exist create one and add it to element
    if (!elementId) {
      elementId = (Math.abs(Math.random()) * 10000000).toFixed(0)
      element.setAttribute(elementIdAttributeName, elementId)
    }

    return {selector:`[${elementIdAttributeName}="${elementId}"]`}
  }
```